src/main/java/com/code_intelligence/jazzer/mutation/mutator/aggregate/BeanSupport.java

```diff
@@ -88,9 +117,11 @@ static Optional<Method[]> findGettersByPropertyNames(
         propertyNames.map(gettersByPropertyName::get).map(Optional::ofNullable), Method[]::new);
   }
 
-  static Optional<Method[]> findGettersByPropertyTypes(Class<?> clazz, Stream<Class<?>> types) {
-    Map<Class<?>, List<Method>> gettersByType =
-        findMethods(clazz, BeanSupport::isGetter).collect(groupingBy(Method::getReturnType));
+  static Optional<Method[]> findGettersByPropertyTypes(
+      Class<?> clazz, AnnotatedType classType, Stream<Type> types) {
+    Map<Type, List<Method>> gettersByType =
+        findMethods(clazz, BeanSupport::isGetter)
+            .collect(groupingBy(m -> resolveReturnType(m, classType)));
```

Comment:
If `m` is not declared in `classType`, which here is the instantiated class of the concrete bean, the resolution doesn't work, as seen in the examples in another comment below.
To fix it, we should probably resolve all necessary generic types untill we reach the superclass where `m` is declared.
The same applies for the return types of the setters.